### PR TITLE
Update `target_feature` syntax

### PR DESCRIPTION
**Pull request by alexcrichton**
This commit updates to the latest nightly's syntax where `#[target_feature =
"+foo"]` is now deprecated in favor of `#[target_feature(enable = "foo")]`.
Additionally `#[target_feature]` can only be applied to `unsafe` functions for
now.

Along the way this removes a few exampels that were just left around and also
disables the `fxsr` modules as that target feature will need to land in upstream
rust-lang/rust first as it's currently unknown to the compiler.

### CODE REVIEW

**Comment by alexcrichton** [comment]
This was like 99% a sed script btw, and if desired I can try to split it up more based on the script

**Comment by gnzlbg** [comment]
Is there a PR for enabling `fxsr` upstream or should I do that?

File: coresimd/src/x86/i586/sse42.rs [hunk 1]
```diff
@@ -186,8 +186,8 @@ pub unsafe fn _mm_cmpistrm(a: __m128i, b: __m128i, imm8: i32) -> __m128i {
 /// #
 /// # fn main() {
 /// #     if cfg_feature_enabled!("sse4.2") {
-/// #         #[target_feature = "+sse4.2"]
-/// #         fn worker() {
+/// #         #[target_feature(enable = "sse4.2")]
+/// #         unsafe fn worker() {
```
Comment (gnzlbg) [inline coresimd/src/x86/i586/sse42.rs]: Is `unsafe` required now? Cool!

File: examples/types.rs [hunk 2]
```diff
@@ -1,37 +0,0 @@
-#![cfg_attr(feature = "strict", deny(warnings))]
-#![feature(target_feature)]
```
Comment (gnzlbg) [inline examples/types.rs]: What happened to `examples/types.rs` and `examples/wat.rs` ?

File: coresimd/src/x86/i586/sse42.rs (hunk 1 above)
Comment (alexcrichton) [reply coresimd/src/x86/i586/sse42.rs]: It is indeed!

File: examples/types.rs (hunk 2 above)
Comment (alexcrichton) [reply examples/types.rs]: Ah these just looked like things that were playing around and not actually legitimate examples (in the sense of things we wanted to maintain over time) so I just deleted them.

Comment (gnzlbg) [reply examples/types.rs]: I also thought this, but never really looked into them.

**Comment by gnzlbg** [comment]
So LGTM, merge once tests pass!

**Comment by alexcrichton** [comment]
Ah looks like rustc also doesn't know about `v7` on arm!

**Pull request merged**